pkg/utils/util.go

```diff
@@ -9,3 +16,66 @@ func Contains(slice []string, value string) bool {
 	}
 	return false
 }
+
+// GetToolchainClusterName produces a name for ToolchainCluster object that is both deterministic and "reasonably unique".
+// The `ordinal` must be greater than 0 if there are multiple ToolchainCluster objects pointing to the same cluster. This
+// needs to be determined by the caller prior to calling this method.
+func GetToolchainClusterName(clusterType, serverAPIEndpoint string, ordinal int) (string, error) {
+	// NOTE: this function is ported from the original add-cluster.sh script to produce the same names during the transition
+	// period to the new operator-based approach to the member registration.
+	// Since add-cluster.sh was a bash script with a long history, the logic is a bit convoluted at places (especially in
+	// handling the numerical suffix).
+
+	// we need to make sure that:
+	// 1) the name is at most 63 characters long
+	// 2) the variable part is (a part of) the cluster hostname
+	// 3) it ends with a digit (supplied by the ordinal param) if it was shortened
+
+	var ordinalLength int
+	if ordinal > 0 {
+		ordinalLength = int(math.Log10(float64(ordinal))) + 1
+	} else {
+		ordinalLength = 0
+	}
+
+	// the name always contains the cluster type, a hypen between the cluster type and the numerical suffix (if needed)
+	// Interestingly, this is computed BEFORE we determine if we need the numerical suffix at all, but that's the logic
+	// in the original script.
+	fixedLength := len(clusterType) + ordinalLength + 1
+
+	maxAllowedClusterHostNameLen := 62 - fixedLength // I think 62 is here, because we might default the ordinal to 1 later on and we could theoretically
```

Comment:
ok, I found also the discussion in the PR where it was added: 
https://github.com/codeready-toolchain/toolchain-cicd/pull/83#discussion_r1135419046

btw, these long names occur only for clusters provisioned via a cluster-bot - in production we don't have a cluster name that would have such a long name, so it's totally OK to change it either to 63 or 60 or anything like that. It won't break any running service. 
The only importance is:
* that the name won't be longer than 63 chars so it can be used as a label
* we keep backward compatibility for the clusters with shorter names (which is the case from production/stage environments)